objc/objc_runtime_darwin.go

```diff
@@ -24,27 +24,41 @@ var (
 
 	objc_msgSend              = purego.Dlsym(objc, "objc_msgSend")
 	objc_msgSendSuper2        = purego.Dlsym(objc, "objc_msgSendSuper2")
-	objc_getClass             = purego.Dlsym(objc, "objc_getClass")
-	objc_getProtocol          = purego.Dlsym(objc, "objc_getProtocol")
-	objc_allocateClassPair    = purego.Dlsym(objc, "objc_allocateClassPair")
-	objc_registerClassPair    = purego.Dlsym(objc, "objc_registerClassPair")
-	sel_registerName          = purego.Dlsym(objc, "sel_registerName")
-	class_getSuperclass       = purego.Dlsym(objc, "class_getSuperclass")
-	class_getInstanceVariable = purego.Dlsym(objc, "class_getInstanceVariable")
-	class_addMethod           = purego.Dlsym(objc, "class_addMethod")
-	class_addIvar             = purego.Dlsym(objc, "class_addIvar")
-	class_addProtocol         = purego.Dlsym(objc, "class_addProtocol")
-	ivar_getOffset            = purego.Dlsym(objc, "ivar_getOffset")
-	object_getClass           = purego.Dlsym(objc, "object_getClass")
+	objc_getClass             func(name string) Class
+	objc_getProtocol          func(name string) *Protocol
+	objc_allocateClassPair    func(super Class, name string, extraBytes uintptr) Class
+	objc_registerClassPair    func(class Class)
+	sel_registerName          func(name string) SEL
+	class_getSuperclass       func(class Class) Class
+	class_getInstanceVariable func(class Class, name string) Ivar
+	class_addMethod           func(class Class, name SEL, imp IMP, types string) bool
+	class_addIvar             func(class Class, name string, size uintptr, alignment uint8, types string) bool
+	class_addProtocol         func(class Class, protocol *Protocol) bool
+	ivar_getOffset            func(ivar Ivar) uintptr
+	object_getClass           func(obj ID) Class
 )
 
+func init() {
+	purego.Func(objc, "object_getClass", &object_getClass)
+	purego.Func(objc, "objc_getClass", &objc_getClass)
+	purego.Func(objc, "objc_getProtocol", &objc_getProtocol)
+	purego.Func(objc, "objc_allocateClassPair", &objc_allocateClassPair)
+	purego.Func(objc, "objc_registerClassPair", &objc_registerClassPair)
+	purego.Func(objc, "sel_registerName", &sel_registerName)
+	purego.Func(objc, "class_getSuperclass", &class_getSuperclass)
+	purego.Func(objc, "class_getInstanceVariable", &class_getInstanceVariable)
+	purego.Func(objc, "class_addMethod", &class_addMethod)
+	purego.Func(objc, "class_addIvar", &class_addIvar)
+	purego.Func(objc, "class_addProtocol", &class_addProtocol)
+	purego.Func(objc, "ivar_getOffset", &ivar_getOffset)
```

Comment:
We should handle errors from `purego.Func`

Reply:
Did you see my question about if `purego.Func` should return an error in the first place?

Reply:
Sorry I read it now. So you don't want not to return an error from Func, right? I think it's OK to panic instead of returning error. This is not Go-conventional in general, but purego is a low-layer API and panicking should be allowed like what `reflect` package does.